herd/tests/instructions/AArch64/A164.litmus

```diff
@@ -0,0 +1,13 @@
+AArch64 A164
+(* CMP shifted register is equivalent to SUBS WZR *)
+(* tests xreg *)
+
+ { 0:X1=1; }
+
+P0;
+  CMP XZR, X1, ASR #1;
+  B.EQ foo;
+  MOV X0, #1;
+  foo: NOP;
+
+forall (0:X1=1 /\ 0:XZR=0 /\ 0:X0=0)
```

Comment:
This test is slightly incorrect as regards printing rules, litmus and herd print registers according to their types. Here it does not harm but it gives me the opportunity to detail the connections  between types and printing.

Implicitly this type is `int` (32bits). Here all registers are used in X form, and hence are meant to contain 64bit integers. This is no big deal because the values in registers are small positive integer. This would be a problem if, for instance 0:X1 would contain 2^32, then it will be printed as zero and not as 4294967296.
Finally to avoid such (infrequent) surprises, I strive to use W registers or to declare appropriate types for printed registers (here `int64_t 0:X1; int64_t 0:X0;`, It does not seem very useful for XZR).